src/Auth0.AuthenticationApi/Tokens/IdTokenValidator.cs

```diff
@@ -33,9 +33,8 @@ internal static async Task AssertTokenMeetsRequirements(this IdTokenRequirements
 
             var token = DecodeToken(rawIDToken);
 
-            // For now we want to support HS256 + ClientSecret as we just had a major release.
-            // TODO: In the next major (v4.0) we should remove this condition as well as Auth0ClientOptions.ClientSecret
```

Comment:
I'm confused with this comment about v4 and the one below for v7. Should we keep supporting client_secret (and hs256) or not?

Reply:
This was a hold-over from borrowing the code from auth0-oidc-client-net and can be ignored.